packages/devtools_app/lib/src/shared/utils.dart

```diff
@@ -297,3 +298,78 @@ Future<void> launchUrlWithErrorHandling(String url) async {
     onError: () => notificationService.push('Unable to open $url.'),
   );
 }
+
+/// A worker that will run [callback] in groups of [chunkSize], when [doWork] is called.
+///
+/// [progressCallback] will be called with 0.0 progress when starting the work and any
+/// time a chunk finishes running, with a value that represents the proportion of
+/// indices that have been completed so far.
+///
+/// This class may be helpful when sets of work need to be done over an array, while
+/// avoiding blocking the UI thread.
+class InterruptableChunkWorker {
+  InterruptableChunkWorker({
+    int chunkSize = 50,
+    required this.callback,
+    required this.progressCallback,
+  }) : _chunkSize = chunkSize;
+
+  final int _chunkSize;
+  int _workId = 0;
+  void Function(int) callback;
+  void Function(double progress) progressCallback;
+
+  final _sw = Stopwatch();
+
+  /// Start doing the chunked work.
+  ///
+  /// [callback] will be called on every index from 0...[length-1], inclusive,
+  /// in chunks of [_chunkSize]
+  ///
+  /// If [doWork] is called again, then [callback] will no longer be called
+  /// on any remaining indices from previous [doWork] calls.
+  ///
+  Future<bool> doWork(
+    int length,
+  ) {
+    final completer = Completer<bool>();
+    final localWorkId = ++_workId;
+    final sw = Stopwatch();
+
+    Function(int i)? doChunkWork;
+
+    doChunkWork = (i) {
```

Comment:
make this a local function directly instead of having a variable to store the function:
```
void doChunkWork(int i) {
  ...
}
```